### PR TITLE
[iOS] Implement ScrollView Orientation

### LINKED ISSUE #13187: [Android][Mac Catalyst]  ScrollView with Orientation Both cannot scroll both directions (missing scrollbar)
### Description

For Android,  You can only slide left and right to display the page, you cannot slide up and down.
For Mac Catalyst， You can only slide slide up and down to display the page, you cannot left and right .
Windows and iOS Work fine.

Windows

Andorid

Mac Catalyst

iOS

### Steps to Reproduce

Example of a GitHub repository https://github.com/gudder/MauiApp2
Put GraphicsView inside ScrollView.

### Link to public reproduction project repository

https://github.com/gudder/MauiApp2

### Version with bug

6.0.312

### Last version that worked well

Unknown/Other

### Affected platforms

Android, macOS

### Affected platform versions

Andorid 12  maccatalyst 14

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

### LINKED ISSUE #13412: Setting ScrollView.Orientation has no effect on iOS
### Description

No matter what value I set for a ScrollView's Orientation property, I can scroll in all directions on iOS.

Simulator Screen Shot - iPhone SE (3rd generation) - 2023-02-17 at 10 16 19

### Steps to Reproduce

1. Create a new .NET MAUI App (I'm targeting .NET 6.0, not sure if that makes a difference)
2. Replace the XAML of the MainPage with
```
<?xml version="1.0" encoding="utf-8" ?>
<ContentPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             x:Class="MyMAUIApp.MainPage">
    <ScrollView Orientation="Vertical" HorizontalOptions="FillAndExpand" VerticalOptions="FillAndExpand" BackgroundColor="Red">
        <BoxView HeightRequest="800" WidthRequest="800" Color="Transparent">
            <BoxView.Background>
                <LinearGradientBrush>
                    <GradientStop Color="Yellow" Offset="0.1" />
                    <GradientStop Color="Green" Offset="1.0" />
                </LinearGradientBrush>
            </BoxView.Background>
        </BoxView>
    </ScrollView>
</ContentPage>
```
3. Remove everything but the constructor in MainPage.xaml.cs, to make sure the project compiles
4. Run the app on an iPhone SE simulator
5. Observe that you can scroll both horizontally and vertically (see screenshot above)

### Link to public reproduction project repository

n/a

### Version with bug

7.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Affected platform versions

iOS 16.2

### Did you find any workaround?

Listen to the `Scrolled` event and make sure it always scrolls to 0, e.g. with
```
HorizontalScrollView.ScrollToAsync(e.ScrollX, 0, false);
```

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by rmarinho**
### Description of Change

Seems there was no specific implementation for ScrollViewOrientation on iOS. So it was always allowing to scroll in both directions, 13412 

The issue is the ContentSize of the UIScrollView is not set on the correct dimension by constraining the size of the scrollview depending on the orientation. 
The implementation of `IContentView.CrossPlatformMeasure` on ScrollView.Impl does try to constrain when measuring using the orientation but the final `DesiredSize` is always the expressed WidthRequested/HeightRequested.

So one way to fix is to take in account the orientation when setting the UIScrollView ContentSize.

### Issues Fixed

Fixes #13412
Fixes #13187

### CODE REVIEW

File: src/Core/src/Platform/iOS/ScrollViewExtensions.cs [hunk 1]
```diff
@@ -40,19 +40,34 @@ public static void UpdateContent(this UIScrollView scrollView, IView? content, I
...
+			var bounds = platformScrollView.GetPlatformViewBounds();
+			var widthConstraint = bounds.Width;
+			var heightConstraint = bounds.Height;
+			var result = scrollView.CrossPlatformMeasure(widthConstraint, heightConstraint);
+			var contentSize = result.AccountForOrientation(widthConstraint, heightConstraint, scrollView);
+			platformScrollView.ContentSize = contentSize;
 		}
 
-		public static void UpdateIsEnabled(this UIScrollView nativeScrollView, IScrollView scrollView)
+		internal static Size AccountForOrientation(this Size size, double widthConstraint, double heightConstraint, IScrollView view)
```
Comment (rmarinho) [inline src/Core/src/Platform/iOS/ScrollViewExtensions.cs:58]: Should this be public now since is going to net8?

**Review by mandel-macaque — APPROVED** [review]

**Comment by rmarinho** [comment]
@hartez I have specially doubts also why do we have 2 methods setting the ContentSize of the UIScrollView.

**Review by hartez — CHANGES_REQUESTED** [review]

File: src/Core/src/Handlers/ScrollView/ScrollViewHandler.iOS.cs [hunk 2]
```diff
@@ -55,15 +55,9 @@ public static void MapContent(IScrollViewHandler handler, IScrollView scrollView
 			UpdateContentView(scrollView, handler);
 		}
 
-		// We don't actually have this mapped because we don't need it, but we can't remove it because it's public
```
[the remark this answers is not in the document]
Comment (rmarinho) [reply src/Core/src/Handlers/ScrollView/ScrollViewHandler.iOS.cs:58]: Ok I will remove this so we might want to also backport this pr.

File: src/Core/src/Handlers/ScrollView/ScrollViewHandler.iOS.cs [hunk 3]
```diff
@@ -78,7 +72,7 @@ public static void MapVerticalScrollBarVisibility(IScrollViewHandler handler, IS
 
 		public static void MapOrientation(IScrollViewHandler handler, IScrollView scrollView)
 		{
-			// Nothing to do here for now, but we might need to make adjustments for FlowDirection when the orientation is set to Horizontal
+			handler.PlatformView?.UpdateOrientation(scrollView);
```
[the remark this answers is not in the document]
Comment (rmarinho) [reply src/Core/src/Handlers/ScrollView/ScrollViewHandler.iOS.cs:81]: I have tried with SetNeedsLayout() but that doesn't work. Doesn't seem to trigger any Measure or Arrange. 
The `UpdateOrieantation` will set the ContentSize.

File: src/Core/src/Platform/iOS/ScrollViewExtensions.cs (hunk 1 above)
Comment (rmarinho) [reply src/Core/src/Platform/iOS/ScrollViewExtensions.cs]: Fixed!

File: src/Core/src/Handlers/ScrollView/ScrollViewHandler.iOS.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (rmarinho) [reply src/Core/src/Handlers/ScrollView/ScrollViewHandler.iOS.cs:81]: I think I figured it out, I need to grab is the child ContainerView we create to host the Content

[the remark this answers is not in the document]
Comment (rmarinho) [reply src/Core/src/Handlers/ScrollView/ScrollViewHandler.iOS.cs]: Should be good now

**Review by hartez — APPROVED** [review]

**Comment by Glorfindel83** [comment]
Thanks for fixing this - what's the easiest way to find out in which MAUI version this will be available? I have the repository downloaded but even with SourceTree it's complicated to get a good overview. (It's not that critical for me that I need to compile my own MAUI, I'm just using the Stable update channel in Visual Studio.)

**Comment by rmarinho** [comment]
it would be in a net8 version, p3 or p4

**Pull request merged**